src/components/RegisterNewUser.jsx

```diff
@@ -1,108 +1,55 @@
-/*
-The form that a new user fills out to register a new account.
-The user provides their name, an email, a password, and confirms
-their password.
-*/
-import React from 'react';
-import $ from 'jquery';
+import React, { PropTypes } from 'react';
 
-// component for registering as a new user
-export default class RegisterNewUser extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      username: '',
-      password: '',
-      confirmPassword: '',
-      passwordMatch: false,
-    };
-    this.updateEmail = this.updateEmail.bind(this);
-    this.updatePassword = this.updatePassword.bind(this);
-    this.matchPasswords = this.matchPasswords.bind(this);
-    this.register = this.register.bind(this);
-  }
-  // updates current state to value of email field
-  updateEmail(event) {
-    this.setState({
-      username: event.target.value,
-    });
-  }
-  // updates current state to value of password field
-  updatePassword(event) {
-    this.setState({
-      password: event.target.value,
-    });
-  }
-  // updates current state to value of confirm password field
-  // and determines whether or not the passwords match
-  matchPasswords(event) {
-    this.setState({
-      confirmPassword: event.target.value,
-    });
-    if (event.target.value === this.state.password) {
-      this.setState({
-        passwordMatch: true,
-      });
-    } else {
-      this.setState({
-        passwordMatch: false,
-      });
-    }
-  }
-  register(event) {
-    // if a password is provided and correctly verified
-    // the new user account information is sent to the user
-    // database and the account is created
-    event.preventDefault();
-    if (this.state.passwordMatch === false) {
-      alert('Passwords do not match, please try again');
-    } else if (this.state.username === '') {
-      alert('An email must be provided');
-    } else {
-      const formData = {
-        username: this.state.username,
-        password: this.state.password,
-      };
+const RegisterNewUser = ({
+  username,
+  password,
+  confirmPassword,
+  updateNewUsername,
+  updateNewPassword,
+  updateConfirmPassword,
+  onRegister,
+}) => (
+  <div>
+    <form id="register" className="form-signin" onSubmit={onRegister}>
+      <h2>Register an account</h2>
+      <br />
+      <input
+        placeholder="Email"
+        type="text"
+        value={username}
+        onChange={updateNewUsername}
+      />
+      <br />
+      <input
+        placeholder="Password"
+        type="password"
+        value={password}
+        onChange={updateNewPassword}
+      />
+      <br />
+      <input
+        placeholder="Confirm Password"
```

Comment:
It doesn't really seem like we're doing anything really with "confirm password" field (ideally, it would do a check between this field and the new password field, and alert if not the same). Let's remove this for now, and tag this as future work, since it's not super crucial to implement for now.